strawberry_django/mutations/resolvers.py

```diff
@@ -366,16 +376,24 @@ def create(
         key_attr=key_attr,
     )
 
-    # Creating the instance directly via create() without full-clean will
-    # raise ugly error messages. To generate user-friendly ones, we want
-    # full-clean() to trigger form-validation style error messages.
-    full_clean_options = full_clean if isinstance(full_clean, dict) else {}
-    if full_clean:
+    try:
+        # Instead of using `get_or_create` shortcut first use `get()`
+        # (which will also raise `DoesNotExist` in case pk `None` value explicitly
+        # passed), if no records found - invoke `full_clean()` and `create`
+        # right after. The idea is to bypass cleaning first to allow getting
+        # an object by e.g. unique fields. In case `full_clean()` invoked before,
+        # it would raise unique constraint validation error
+        instance = model._default_manager.get(**create_kwargs)
+    except model.DoesNotExist:
+        # Creating the instance directly via create() without full-clean will
+        # raise ugly error messages. To generate user-friendly ones, we want
+        # full-clean() to trigger form-validation style error messages.
+        full_clean_options = full_clean if isinstance(full_clean, dict) else {}
         dummy_instance.full_clean(**full_clean_options)  # type: ignore
```

Comment:
Instead of using `get_or_create` shortcut I finally came up with a decision to use `get` and `create` separately, but invoke `full_clean` only after `get()` is invoked as `full_clean` may raise unique constraint violation exception (e.g. in use case described in #360 )